truthbeam/factory_test.go

```diff
@@ -0,0 +1,101 @@
+package truthbeam
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/config/configcompression"
+	"go.opentelemetry.io/collector/config/confighttp"
+)
+
+func TestCreateDefaultConfig(t *testing.T) {
+	factory := NewFactory()
+	config := factory.CreateDefaultConfig()
+
+	require.NotNil(t, config, "Config should not be nil")
+
+	cfg, ok := config.(*Config)
```

Comment:
This is smart and has some Go concepts behind the scenes. In a first glance one could think it is redundant with the previous test (checking `nil`). But this test is essentially doing a type validation, which is interesting.

Maybe including a comment to make it clear could help.